### PR TITLE
TiFlash: Update max-replicas settings before enable-placement-rules

### PR DESCRIPTION
**Pull request by birdstorm**
### What problem does this PR solve? 

Settings of max-replicas may be overwritten by enable-placement-rules if it was set after enable-placement-rules is set to true.

Manually tested.

### What is changed and how it works?

Move the setting of max-replicas before enable-placement-rules.

Also, add low-space-ratio check.

### Check List 

Tests 

 - Manual test

Code changes

 - Has exported function/method change
 - Has interface methods change

Side effects

 - Possible performance regression

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: components/playground/instance/tiflash.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"time"
 
 	"github.com/pingcap/errors"
-	"github.com/pingcap/tiup/components/playground/api"
+	"github.com/pingcap/tiup/pkg/cluster/api"
```
Comment (JaySon-Huang) [inline components/playground/instance/tiflash.go:29]: So "playground" is somehow relied on "cluster"?

File: components/playground/instance/tiflash.go [hunk 2]
```diff
@@ -96,7 +104,28 @@ func (inst *TiFlashInstance) Start(ctx context.Context, version v0manifest.Versi
 
 	// Wait for PD
 	pdClient := api.NewPDClient(endpoints, 10*time.Second, nil)
-	enablePlacementRules, err := json.Marshal(replicateConfig{
+	// set low-space-ratio to 1 to avoid low disk space
+	lowSpaceRatio, err := json.Marshal(scheduleConfig{
+		LowSpaceRatio: 1,
```
Comment (JaySon-Huang) [inline components/playground/instance/tiflash.go:109]: In PD's [document](Maximum value: less than 1), it is "less than 1", but not "less than or equal 1".
Maybe set it to 0.99?

File: components/playground/instance/tiflash.go [hunk 3]
```diff
@@ -96,7 +104,28 @@ func (inst *TiFlashInstance) Start(ctx context.Context, version v0manifest.Versi
...
+	})
+	if err != nil {
+		return nil
```
Comment (JaySon-Huang) [inline components/playground/instance/tiflash.go:112]: return err ?

File: components/playground/instance/tiflash.go [hunk 4]
```diff
@@ -96,7 +104,28 @@ func (inst *TiFlashInstance) Start(ctx context.Context, version v0manifest.Versi
...
+	}
+	if err = pdClient.UpdateScheduleConfig(bytes.NewBuffer(lowSpaceRatio)); err != nil {
+		return err
+	}
+	// Update maxReplicas before placement rules so that it would not be overwritten
+	maxReplicas, err := json.Marshal(replicateMaxReplicaConfig{
+		MaxReplicas: 1,
+	})
+	if err != nil {
+		return nil
```
Comment (JaySon-Huang) [inline components/playground/instance/tiflash.go:122]: ditto

File: pkg/cluster/api/pdapi.go [hunk 5]
```diff
@@ -537,23 +538,20 @@ func (pc *PDClient) DelPD(name string, retryOpt *clusterutil.RetryOption) error
 	return nil
 }
 
-// IsTombStone check if the node is Tombstone.
-// The host parameter should be in format of IP:Port, that matches store's address
-func (pc *PDClient) IsTombStone(host string) (bool, error) {
+func (pc *PDClient) isState(host string, state metapb.StoreState) (bool, error) {
```
Comment (JaySon-Huang) [inline pkg/cluster/api/pdapi.go:541]: Maybe `isSameState` is better

File: components/playground/instance/tiflash.go (hunk 2 above)
Comment (birdstorm) [reply components/playground/instance/tiflash.go:109]: sure

File: components/playground/instance/tiflash.go (hunk 3 above)
Comment (birdstorm) [reply components/playground/instance/tiflash.go:112]: yes. i will fix it.

File: pkg/cluster/api/pdapi.go (hunk 5 above)
Comment (birdstorm) [reply pkg/cluster/api/pdapi.go:541]: ok

File: components/playground/instance/tiflash.go (hunk 1 above)
Comment (birdstorm) [reply components/playground/instance/tiflash.go:29]: the api of PDClient is copied from cluster. I think I will move them to a common package afterwards.

Comment (birdstorm) [reply components/playground/instance/tiflash.go:29]: @JaySon-Huang It actually relies on `pkg`, `cluster` is a sub-directory of `pkg`, different from module `cluster`

**Comment by bearHunting** [comment]
very effecient! Actually I found this problem and report to authority. Glad to help!

**Pull request merged**